testplan/runners/pools/base.py

```diff
@@ -620,19 +628,25 @@ def starting(self):
         """Starting the pool and workers."""
         # TODO do we need a lock here?
         self._conn.start()
+
+        for worker in self._workers:
+            # reset worker (if any) status
+            worker.status.change(ResourceStatus.STARTING)
+
+        self._exit_loop = False
+        super(Pool, self).starting()  # start the loop & monitor
+
         if not self._workers:
             self._add_workers()
         self._start_workers()
 
         if self._workers.start_exceptions:
             for msg in self._workers.start_exceptions.values():
                 self.logger.error(msg)
-            self._workers.stop()
+            self.stop()
```

Comment:
I think should call `self.abort()` - stop is for stopping under normal conditions, abort is for error conditions.